Dockerfile

```diff
@@ -0,0 +1,106 @@
+
+# Installs system dependencies required to run Ruby
+RUN set -eux; \
+  DEBIAN_FRONTEND=noninteractive apt-get update; \
+  DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  bzip2 \
+  ca-certificates \
+  libgmp-dev \
+  libssl-dev \
+  libyaml-dev \
+  procps \
+  zlib1g-dev \
+  ; \
+  rm -rf /var/lib/apt/lists/*; \
+  # skip installing gem documentation
+  mkdir -p /usr/local/etc; \
+  { \
+  echo 'install: --no-document'; \
+  echo 'update: --no-document'; \
+  } >> /usr/local/etc/gemrc; \
+  # Installs the dependencies required to build Ruby
+  savedAptMark="$(apt-mark showmanual)"; \
+  DEBIAN_FRONTEND=noninteractive apt-get update; \
+  DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  git \
+  dpkg-dev \
+  libgdbm-dev \
+  ruby \
+  autoconf \
+  g++ \
+  gcc \
+  libbz2-dev \
+  libgdbm-compat-dev \
+  libglib2.0-dev \
+  libncurses-dev \
+  libxml2-dev \
+  libxslt-dev \
+  make \
+  wget \
+  xz-utils \
+  ; \
+  rm -rf /var/lib/apt/lists/*; \
+  # Install Rust to build Ruby with YJIT
+  rustArch=; \
+  dpkgArch="$(dpkg --print-architecture)"; \
+  case "$dpkgArch" in \
+  'amd64') rustArch='x86_64-unknown-linux-gnu'; rustupUrl='https://static.rust-lang.org/rustup/archive/1.26.0/x86_64-unknown-linux-gnu/rustup-init'; rustupSha256='0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db' ;; \
+  'arm64') rustArch='aarch64-unknown-linux-gnu'; rustupUrl='https://static.rust-lang.org/rustup/archive/1.26.0/aarch64-unknown-linux-gnu/rustup-init'; rustupSha256='673e336c81c65e6b16dcdede33f4cc9ed0f08bde1dbe7a935f113605292dc800' ;; \
+  *) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
+  esac; \
+  mkdir -p /tmp/rust; \
+  wget -O /tmp/rust/rustup-init "$rustupUrl"; \
+  echo "$rustupSha256 */tmp/rust/rustup-init" | sha256sum --check --strict; \
+  chmod +x /tmp/rust/rustup-init; \
+  export RUSTUP_HOME='/tmp/rust/rustup' CARGO_HOME='/tmp/rust/cargo'; \
+  export PATH="$CARGO_HOME/bin:$PATH"; \
+  /tmp/rust/rustup-init -y --no-modify-path --profile minimal --default-toolchain '1.77.0' --default-host "$rustArch"; \
+  rustc --version; \
+  cargo --version; \
+  # Install Ruby with https://github.com/rbenv/ruby-build
+  mkdir /src; \
+  git -C /src clone https://github.com/rbenv/ruby-build.git; \
+  cd /src/ruby-build && ./install.sh; \
+  cd / && rm -fr /src; \
+  export RUSTUP_HOME='/tmp/rust/rustup' CARGO_HOME='/tmp/rust/cargo';\
+  export PATH="$CARGO_HOME/bin:$PATH"; \
+  CONFIGURE_OPTS="--disable-install-doc --enable-yjit" ruby-build ${RUBY_VERSION} /usr/local; \
+  # Cleanup build dependencies
+  rm -rf /tmp/rust; \
+  rm -rf /usr/local/bin/ruby-build; \
+  apt-mark auto '.*' > /dev/null; \
+  apt-mark manual $savedAptMark > /dev/null; \
+  find /usr/local -type f -executable -not \( -name '*tkinter*' \) -exec ldd '{}' ';' \
+  | awk '/=>/ { so = $(NF-1); if (index(so, "/usr/local/") == 1) { next }; gsub("^/(usr/)?", "", so); printf "*%s\n", so }' \
```

Comment:
Nitpick: I would also indent these just because they are somewhat related to the previous command.